### PR TITLE
Add recipe for generating dummy soil data

### PR DESCRIPTION
**Pull request by jacobcook1995**
# Description

This pull request adds a recipe for generating the dummy soil data file to the data recipes section of the docs. This is a way of storing the code to generate the dummy data files in the medium term, and will likely be superseded by instructions on how to generate appropriate data files containing real data eventually.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [N/A] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: docs/source/data_recipes/soil_dummy.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+bulk_density_values = [[generate_BD_values(x, y) for y in y_coords] for x in x_coords]
+percent_clay_values = [[generate_clay_values(x, y) for y in y_coords] for x in x_coords]
+lmwc_values = [[generate_lmwc_values(x, y) for y in y_coords] for x in x_coords]
+maom_values = [[generate_maom_values(x, y) for y in y_coords] for x in x_coords]
+microbial_C_values = [
+    [generate_microbial_C_values(x, y) for y in y_coords] for x in x_coords
+]
+
+# List of displacements (applies to both x and y)
+disps = [0, 90, 180, 270, 360, 450, 540, 630, 720]
```
Comment (vgro) [inline docs/source/data_recipes/soil_dummy.md:71]: What do you mean by `displacement`? I assume this has to do with the grid? Maybe add a sentence (or include in the intro when you write the section.)

File: docs/source/data_recipes/soil_dummy.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    """Function to generate a reasonable range of maom values."""
+    return 1.0 + 2.0 * (x * y) / (64)
+
+
+def generate_microbial_C_values(x: float, y: float) -> float:
+    """Function to generate a reasonable range of microbial C values."""
+    return 0.0015 + 0.0035 * (x * y) / (64)
+
+
+# Generate range of x and y values
```
[the remark this answers is not in the document]
Comment (jacobcook1995) [reply docs/source/data_recipes/soil_dummy.md:56]: Ahh yeah that's a good point they are basically cell numbers (in the `x` and `y` directions, respectively), but I don't actually say that anywhere

File: docs/source/data_recipes/soil_dummy.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        soil_c_pool_maom=(["x", "y"], maom_values),
+        soil_c_pool_microbe=(["x", "y"], microbial_C_values),
+    ),
+    coords=dict(
+        x=(["x"], disps),
+        y=(["y"], disps),
+    ),
+    attrs=dict(description="Soil data for dummy Virtual Rainforest model."),
+)
+```
```
Comment (vgro) [inline docs/source/data_recipes/soil_dummy.md:110]: It would be nice to show the `dummy_soil_data` here

**Comment by jacobcook1995** [comment]
Ahh sorry @vgro, just realised that I made the pull request with the remote branch a few commits behind my local branch. It doesn't affect any of the comments you made (which I still need to address), but the TODO's at the top of the file are addressed

**Comment by vgro** [comment]
> Ahh sorry @vgro, just realised that I made the pull request with the remote branch a few commits behind my local branch. It doesn't affect any of the comments you made (which I still need to address), but the TODO's at the top of the file are addressed

No worries. I think it looks overall good, just add a bit more text to explain the steps and variables, then I'm happy for you to merge.

File: docs/source/data_recipes/soil_dummy.md (hunk 1 above)
Comment (jacobcook1995) [reply docs/source/data_recipes/soil_dummy.md:92]: These are the x and y values associated with each cell so (1,1) => (0m, 0m), and (3,5) => (180m, 360m).

Now I think about this do they need to have an offset? i.e. should the list start at 45m (or -45m)?

Comment (jacobcook1995) [reply docs/source/data_recipes/soil_dummy.md:97]: Ahh that makes sense!

**Review by vgro — APPROVED** [review]
LGTM :-)

**Pull request merged**